### PR TITLE
Make sure footnote link target can be properly scrolled into view

### LINKED ISSUE #5788: Offset viewport where footnotes jump to
Offset viewport where footnotes jump to (currently top section hidden behind nav)

Example URL:
- https://foundation.mozilla.org/en/data-futures-lab/data-for-empowerment/what-helps-the-ecosystem-for-needs-and-support/#footnote-1
- click footnote's `[1]` to go to the footnote in the content.
- click the footnote link in the content
- footnote ends up underneath the `.article-navbar` div

### PR DESCRIPTION
**Pull request by mmmavis**
Closes #5788

Sample test page: https://foundation-s-issue-5788-d5en16.herokuapp.com/publication-page-with-child-article-pages/fixed-title-article-page/

### CODE REVIEW

**Review by sabrinang — CHANGES_REQUESTED** [review]
- Great to have the nav stop before the footnotes to have it not block them below
- I like seeing the scroll that takes you back up to where the note is right below the nav that isn't blocked anymore when you click the bottom footnote 
- When you click a footnote in the body, can we also see the scroll down to where the footnote is below?

**Comment by mmmavis** [comment]
@sabrinang PR updated! I assume that we would want smooth scrolling over a sudden jump kinda scrolling for anchors so I've enabled smooth scrolling on all pages on the foundation site.

**Review by sabrinang — APPROVED** [review]
ah yes smooth scrolling both ways, feels great, thank you!

File: source/js/foundation/template-js-handler/article-footnote-link.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      event.preventDefault();
+
+      const target = document.querySelector(link.getAttribute(`href`));
+      const offset = siteNav.scrollHeight + summaryNav.scrollHeight;
+      const top =
+        target.getBoundingClientRect().top + window.pageYOffset - offset;
+
+      window.scrollTo({
+        top: top,
+        behavior: `smooth`,
```
Comment (KalobTaulien) [inline source/js/foundation/template-js-handler/article-footnote-link.js:24]: There's a behaviour property on this?! Amazing!

**Review by KalobTaulien — APPROVED** [review]

**Pull request merged**